utils/endOfLifeDateApi/index.js

```diff
@@ -0,0 +1,78 @@
+const END_OF_LIFE_DATE_BASE_URL = "https://endoflife.date/api";
+const DEFAULT_OPTIONS = {
+  method: "GET",
+  headers: {
+    Accept: "application/json",
+  },
+};
+
+/**
+ * @typedef {Object} Cycle
+ * @property {number|string} cycle - Release cycle
+ * @property {string} releaseDate - Release date for the first release in this cycle
+ * @property {string} eol - End-of-life date for this release cycle
+ * @property {string} latest - Latest release in this cycle
+ * @property {string|null} link - Link to the changelog for the latest release, if available
+ * @property {string|boolean|null} lts - Whether this release cycle has long-term support (LTS).
+ *                                Can be a date instead if the release enters LTS on a
+ *                                given date.
+ * @property {string|boolean|null} support - Whether this release cycle has active support
+ * @property {string|boolean|null} discontinued - Whether this cycle is now discontinued
+ */
+
+/**
+ * Provides a way to interact with the endoflife.data API.
+ */
+export class EndOfLifeDateApiClient {
+  /**
+   * Gets information for all cycles of a given product.
+   * @param {string} product
+   * @returns {Promise<Cycle[]>}
+   */
+  async getAllCycles(product) {
+    const url = `${END_OF_LIFE_DATE_BASE_URL}/${product}.json`;
+
+    const response = await fetch(url, DEFAULT_OPTIONS);
+    return await response.json();
+  }
+
+  /**
+   * Gets information about a specific cycle of a given product.
+   * @param {string} product
+   * @param {number|string} cycle 
+   * @returns {Promise<Cycle>}
+   */
+  async getCycle(product, cycle) {
+    const url = `${END_OF_LIFE_DATE_BASE_URL}/${product}/${cycle}.json`;
+
+    const response = await fetch(url, DEFAULT_OPTIONS);
```

Comment:
Same here, some error handling would be beneficial

Reply:
Similarly to the other function, except it will return `null` instead of an array.